hydra-node/src/Hydra/Chain/Direct/State.hs

```diff
@@ -122,15 +128,20 @@ getKnownUTxO OnChainHeadState{stateMachine} =
   case stateMachine of
     Idle{} ->
       mempty
-    Initialized{initialThreadOutput, initialInitials, initialCommits} ->
+    Initialized{initialThreadOutput = InitialThreadOutput{initialThreadUTxO}, initialInitials, initialCommits} ->
       UTxO $
         Map.fromList $
-          take2Of4 initialThreadOutput : (take2Of3 <$> (initialInitials <> initialCommits))
-    Open{openThreadOutput = (i, o, _, _)} ->
+          take2Of3 initialThreadUTxO : (take2Of3 <$> (initialInitials <> initialCommits))
+    Open{openThreadOutput = OpenThreadOutput{openThreadUTxO = (i, o, _)}} ->
       UTxO.singleton (i, o)
-    Closed{closedThreadOutput = (i, o, _, _)} ->
+    Closed{closedThreadOutput = ClosedThreadOutput{closedThreadUTxO = (i, o, _)}} ->
       UTxO.singleton (i, o)
 
+getContestationDeadline :: OnChainHeadState 'StClosed -> POSIXTime
+getContestationDeadline
+  OnChainHeadState{stateMachine = Closed{closedThreadOutput = ClosedThreadOutput{closedContestationDeadline}}} =
+    closedContestationDeadline
```

Comment:
This is slightly side-stepping the "non-introspectability" of the `OnChainHeadState`, but it's fine given the direction this is going.